client/extensions/woocommerce/app/dashboard/required-plugins-install-view.js

```diff
@@ -94,6 +95,14 @@ class RequiredPluginsInstallView extends Component {
 				this.props.installPlugin( site.ID, wporgPlugin );
 				return;
 			}
+			if ( ! plugin.active ) {
+				if ( -1 < this.state.activatingPlugins.indexOf( slug ) ) {
+					return;
+				}
+				const wporgPlugin = getPlugin( wporg, slug );
+				this.setState( { activatingPlugins: [ ...this.state.activatingPlugins, slug ] } );
+				this.props.activatePlugin( site.ID, { ...wporgPlugin, id: plugin.id } );
```

Comment:
Do we need another `return;` here?

I think in the event the last plugin to check only needs activation, we'd jump right to `setFinishedInstallOfRequiredPlugins()` instead of waiting for the activation request to finish.